client/transport.c

```diff
@@ -353,12 +353,12 @@ static int do_request(int type, xtime_t start_time, xtime_t end_time, void *data
 	expire_time.tv_sec = actual_time + REQUEST_WAIT;
 
 	while(!reply_rcvd){
-		if(pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time)) {
+		if((ret = pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time))) {
 			last_reply_id = reply_id_private;
 			reply_data = NULL;
 			reply_callback = NULL;
-			if(errno != EAGAIN || errno != ETIMEDOUT) {
-				xdag_err("pthread_cond_timedwait failed [function: do_request]");
+			if(ret != EAGAIN || ret != ETIMEDOUT) {
```

Comment:
`ret != EAGAIN || ret != ETIMEDOUT` is always true
You should use `ret != EAGAIN && ret != ETIMEDOUT`

Reply:
wow, yes fixed!